Products/ZenCollector/services/config.py

```diff
@@ -237,16 +237,11 @@ def remote_getDeviceNames(self, options=None):
     def _getDevices(self, deviceNames=None, deviceFilter=None):
 
         if not deviceNames:
-            devices = filter(deviceFilter, self._prefs.devices())
-        else:
-            devices = []
-            for name in deviceNames:
-                device = self.dmd.Devices.findDeviceByIdExact(name)
-                if not device:
-                    continue
-                else:
-                    if deviceFilter(device):
-                        devices.append(device)
+            return filter(deviceFilter, self._prefs.devices())
+        devices = []
+        for device in self._prefs.devices():
```

Comment:
"The performance of a broken system is zero"
And since the catalog might be utterly broken or completely unresponsive, it might not return EVER.  This causes data collection issues and silent failures.  Been there and had to pick up the pieces MULTIPLE times.